packages/x-date-pickers-pro/src/SingleInputDateRangeField/SingleInputDateRangeField.tsx

```diff
@@ -45,7 +46,7 @@ const SingleInputDateRangeField = React.forwardRef(function SingleInputDateRange
     <Input
       ref={ref}
       {...fieldProps}
-      inputProps={{ ...fieldProps.inputProps, ref: inputRef, onPaste, inputMode }}
+      inputProps={{ ...fieldProps.inputProps, ref: inputRef, onPaste, inputMode, readOnly }}
```

Comment:
Single input range fields also do not propagate `readonly` prop correctly, because of `TextField` API. 😢